src/recommendation/src/main/python/RecommendationEvaluator.py

```diff
@@ -0,0 +1,145 @@
+                   typeConverter=TypeConverters.toString)
+
+  rawPredictionCol = Param(Params._dummy(), "rawPredictionCol",
+                           """rawPredictionCol""",
+                           typeConverter=TypeConverters.toString)
+
+  k = Param(Params._dummy(), "k",
+            """k""",
+            typeConverter=TypeConverters.toInt)
+  saveAll = Param(Params._dummy(), "saveAll",
+                  """saveAll""",
+                  typeConverter=TypeConverters.toBoolean)
+
+  nItems = Param(Params._dummy(), "nItems",
+                 """number of items""",
+                 typeConverter=TypeConverters.toInt)
+
+  @keyword_only
+  def __init__(self, rawPredictionCol="rawPrediction", labelCol="label", metricName="ndcgAt", k=3, saveAll=False):
+    """
+    __init__(self, rawPredictionCol="rawPrediction", labelCol="label", \
+             metricName="ndcgAt")
+    """
+    super(RecommendationEvaluator, self).__init__()
+    self._java_obj = self._new_java_obj(
+      "com.microsoft.ml.spark.RecommendationEvaluator", self.uid)
+    self._setDefault(metricName="ndcgAt")
+    self._setDefault(k=3)
+    self._setDefault(saveAll=False)
+    kwargs = self._input_kwargs
+    self._set(**kwargs)
+
+  def setRawPredictionCol(self, value):
+    """
+    Sets the value of :py:attr:`rawPredictionCol`.
+    """
+    return self._set(rawPredictionCol=value)
+
+  def getRawPredictionCol(self):
+    """
+    Gets the value of rawPredictionCol or its default value.
+    """
+    return self.getOrDefault(self.rawPredictionCol)
+
+  def setMetricName(self, value):
+    """
+    Sets the value of :py:attr:`metricName`.
+    """
+    return self._set(metricName=value)
+
+  def getMetricName(self):
+    """
+    Gets the value of metricName or its default value.
+    """
+    return self.getOrDefault(self.metricName)
+
+  def setK(self, value):
+    """
+    Sets the value of :py:attr:`metricName`.
+    """
+    return self._set(k=value)
+
+  def getK(self):
+    """
+    Gets the value of metricName or its default value.
+    """
+    return self.getOrDefault(self.k)
+
+  def setSaveAll(self, value):
+    """
+    Sets the value of :py:attr:`saveAll`.
+    """
+
+    return self._set(saveAll=value)
+
+  def getSaveAll(self):
+    """
+    Gets the value of metricName or its default value.
+    """
+    return self.getOrDefault(self.saveAll)
+
+  def setNumberItems(self, value):
+    """
+    Sets the value of :py:attr:`nItems`.
+    """
+    return self._set(nItems=value)
+
+  @keyword_only
+  def setParams(self, rawPredictionCol="rawPrediction", labelCol="label", metricName="ndcgAt"):
+    """
+    setParams(self, rawPredictionCol="rawPrediction", labelCol="label", \
+              metricName="ndcgAt")
+    Sets params for binary classification evaluator.
+    """
+    kwargs = self._input_kwargs
+    return self._set(**kwargs)
+  
+  def getMetricsList(self)
+    metrics = self._call_java_("getMetricsList).toString()
```

Comment:
looks like you are missing a closing quote. also py4J should handle this conversion if you use java objects instead of ListBuffers